torch/lib/THC/generic/THCTensorMath.cu

```diff
@@ -376,6 +376,28 @@ void THCTensor_(diag)(THCState *state, THCTensor *self_, THCTensor *src_, long k
   THCudaCheck(cudaGetLastError());
 }
 
+void THCTensor_(eye)(THCState *state, THCTensor *self_, long n, long m)
+{
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, self_));
+  THArgCheck(n > 0, 1, "invalid argument");
+
+  if(m <= 0)
+    m = n;
+
+  THCTensor_(resize2d)(state, self_, n, m);
+  THCTensor_(zero)(state, self_);
+
+  long sz = THMin(n, m);
+  long stride = THCTensor_(stride)(state, self_, 0) +
```

Comment:
Also, I suppose you meant `long stride = THCTensor_(size)(state, self_, 1) + 1`?
For reference, here is the [CPU implementation of `eye`](https://github.com/pytorch/pytorch/blob/master/torch/lib/TH/generic/THTensorMath.c#L1994), which uses a similar striding.